pylti1p3/contrib/flask/request.py

```diff
@@ -0,0 +1,24 @@
+from pylti1p3.request import Request
+
+
+class FlaskRequest(Request):
+    _request = None
+    _cookies = []
+
+    def __init__(self, request_kwargs, cookies, session, is_secure):
```

Comment:
Is it possible to pass here only one `request` object instead of `request_kwargs`, `cookies`, `is_secure` ?
I may suggest to operate with this request param in all methods to get all necessary values as it already done for django: https://github.com/dmitry-viskov/pylti1.3/blob/master/pylti1p3/contrib/django/request.py#L8

Reply:
Request object is constructed because of Flask architecture. 

Session is not passed in `request` but you access it as global object 
`class flask.session` https://flask.palletsprojects.com/en/1.1.x/api/#flask.session 

Cookies would be accessible from `class flask.request` as well as information is request secure https://flask.palletsprojects.com/en/1.1.x/quickstart/#cookies

As for parameters I think most people use `marshmallow` or `webargs` etc that results in getting those params from endpoint argument.

So this way (composing "Request" by ourselves) we can use whatever lib we want to process those data before calling lib.

I'm sure that is would be even possible to create middleware or sth similiar that would work out of the box even bypassing need for creation of endpoint but it's much more work than cusomizable "Request".

Reply:
hi @nomysz 

I may suggest such solution:

```
from pylti1p3.request import Request
from flask import request, session


class FlaskRequest(Request):
    _cookies = None
    _session = None
    _request_data = None
    _request_is_secure = None


    def __init__(self, cookies=None, session_data=None, request_data=None, request_is_secure=None):
        self._cookies = request.cookies if cookies is None or cookies
        self._session = session if session_data is None or session_data
        self._request_is_secure = request.is_secure() if request_is_secure is None else request_is_secure
        if request_data:
            self._request_data = request_data

    def get_param(self, key):
        if self._request_data:
            return self._request_data.get(key)
        else:
            if request.method == 'GET':
                return request.args.get(key, None)
            else:
                return request.form.get(key, None)

    def get_cookie(self, key):
        return self._cookies.get(key)

    def is_secure(self):
        return self._request_is_secure
```

here we use Flask `request`, `session` and etc objects by default but anytime may pass custom params from `webargs` and other request objects. What do you think?

Reply:
Done